src/__tests__/index.tsx

```diff
@@ -132,47 +165,106 @@ export default () => leftPad("Neat demo!", 12, '!')
   // this test ensures that *not* passing leftPad as a global here
   // will work because I didn't externalize the left-pad module
   const Component = getMDXComponent(result.code)
-  render(
-    <MDXProvider>
-      <Component />
-    </MDXProvider>,
-  )
+  render(<Component />)
 })
 
 test('gives a handy error when the entry imports a module that cannot be found', async () => {
   const mdxSource = `
 import Demo from './demo'
 
 <Demo />
-`.trim()
+  `.trim()
 
   const error = (await bundleMDX(mdxSource, {
     files: {},
   }).catch(e => e)) as Error
 
-  expect(error.message).toMatchInlineSnapshot(
-    `"Could not resolve './demo' from the entry MDX file"`,
-  )
+  expect(error.message).toMatchInlineSnapshot(`
+    "Build failed with 1 error:
+    __mdx_bundler_fake_dir__/index.mdx:3:17: error: [inMemory] Could not resolve \\"./demo\\" in the entry MDX file."
+  `)
 })
 
 test('gives a handy error when importing a module that cannot be found', async () => {
   const mdxSource = `
 import Demo from './demo'
 
 <Demo />
-`.trim()
+  `.trim()
 
   const error = (await bundleMDX(mdxSource, {
     files: {
       './demo.tsx': `import './blah-blah'`,
     },
   }).catch(e => e)) as Error
 
-  expect(error.message).toMatchInlineSnapshot(
-    `"Could not resolve './blah-blah' from 'demo.tsx'"`,
-  )
+  expect(error.message).toMatchInlineSnapshot(`
+    "Build failed with 1 error:
+    __mdx_bundler_fake_dir__/demo.tsx:1:7: error: [inMemory] Could not resolve \\"./blah-blah\\" in \\"./demo.tsx\\""
+  `)
+})
+
+test('gives a handy error when a file of an unsupported type is provided', async () => {
+  const mdxSource = `
+import Demo from './demo.blah'
+
+<Demo />
+  `.trim()
+
+  const error = (await bundleMDX(mdxSource, {
+    files: {
+      './demo.blah': `what even is this?`,
+    },
+  }).catch(e => e)) as Error
+
+  expect(error.message).toMatchInlineSnapshot(`
+    "Build failed with 1 error:
+    __mdx_bundler_fake_dir__/index.mdx:3:17: error: [JavaScript plugins] Invalid loader: \\"blah\\" (valid: js, jsx, ts, tsx, css, json, text, base64, dataurl, file, binary)"
```

Comment:
I noticed that `.mdx` is not listed as a valid extension by esbuild. Should that work? Does it actually work but it’s not listed by esbuild?